src/dash-playlist-loader.js

```diff
@@ -738,65 +747,63 @@ export default class DashPlaylistLoader extends EventTarget {
         return this.trigger('error');
       }
 
+      // xml is the same do nothing.
+      if (req.responseText === this.masterPlaylistLoader_.masterXml_) {
+        return;
+      }
+
       this.masterPlaylistLoader_.masterXml_ = req.responseText;
+      this.handleMaster_();
 
       // This will filter out updated sidx info from the mapping
       this.masterPlaylistLoader_.sidxMapping_ = filterChangedSidxMappings(
-        this.masterPlaylistLoader_.masterXml_,
-        this.masterPlaylistLoader_.srcUrl,
-        this.masterPlaylistLoader_.clientOffset_,
+        this.masterPlaylistLoader_.master,
         this.masterPlaylistLoader_.sidxMapping_
       );
-
-      const master = parseMasterXml({
-        masterXml: this.masterPlaylistLoader_.masterXml_,
-        srcUrl: this.masterPlaylistLoader_.srcUrl,
-        clientOffset: this.masterPlaylistLoader_.clientOffset_,
-        sidxMapping: this.masterPlaylistLoader_.sidxMapping_
-      });
-      const updatedMaster = updateMaster(this.master, master);
       const currentSidxInfo = this.media().sidx;
 
-      if (updatedMaster) {
-        if (currentSidxInfo) {
-          const sidxKey = generateSidxKey(currentSidxInfo);
+      if (this.media_) {
+        this.media_ = this.master.playlists[this.media_.id];
+      }
 
-          // the sidx was updated, so the previous mapping was removed
-          if (!this.masterPlaylistLoader_.sidxMapping_[sidxKey]) {
-            const playlist = this.media();
+      this.updateMinimumUpdatePeriodTimeout_();
+      // current sidx not updated
```

Comment:
For this and below, would it make more sense to have a `getSidxInfo` function that handles the remainder?

Reply:
This is handled in the next pull request.